src/grandorgue/updater/GOUpdateChecker.h

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2006 Milan Digital Audio LLC
+ * Copyright 2009-2024 GrandOrgue contributors (see AUTHORS)
+ * License GPL-2.0 or later
+ * (https://www.gnu.org/licenses/old-licenses/gpl-2.0.html).
+ */
+
+#ifndef GOUPDATECHECKER_H
+#define GOUPDATECHECKER_H
+
+#include "threading/GOThread.h"
+#include <stdexcept>
+#include <string>
+#include <wx/wx.h>
+
+struct GOUpdateChecker {
+  enum class CheckReason { STARTUP, USER_REQUEST };
+
+  struct ReleaseMetadata {
+    wxString version;
+    wxString changelog;
+  };
+
+  struct Result {
+    bool successful{};
+    ReleaseMetadata latestRelease;
+    bool updateAvailable{};
+    wxString errorMessage;
+    CheckReason checkReason{};
+
+    static Result error(const wxString &message) {
+      Result r;
+      r.successful = false;
+      r.errorMessage = message;
+      return r;
+    }
+
+    static Result success(ReleaseMetadata release, bool updateAvailable) {
+      Result r;
+      r.successful = true;
+      r.latestRelease = std::move(release);
+      r.updateAvailable = updateAvailable;
+      return r;
+    }
+
+    Result &withCheckReason(CheckReason reason) {
+      this->checkReason = reason;
+      return *this;
+    }
+  };
+
+  class CompletionEvent : public wxEvent {
+  private:
+    Result m_result;
+
+  public:
+    CompletionEvent(wxEventType eventType, Result result)
```

Comment:
The result is copied to a field, making at least one copy inevitable. Using 'Result' instead of 'const Result&' in the signature suggests that a copy is made. To prevent additional copies, 'std::move' is utilized.